pages/voorgenomen-huwelijk/getuigen/index.tsx

```diff
@@ -27,74 +29,75 @@ import {
 } from "../../../src/components";
 import { PageFooterTemplate } from "../../../src/components/huwelijksplanner/PageFooterTemplate";
 import { PageHeaderTemplate } from "../../../src/components/huwelijksplanner/PageHeaderTemplate";
-import { exampleState } from "../../../src/data/huwelijksplanner-state";
+import { MarriageOptionsContext } from "../../../src/context/MarriageOptionsContext";
+import { HuwelijkService } from "../../../src/generated";
 
 export const getServerSideProps = async ({ locale }: { locale: string }) => ({
   props: {
     ...(await serverSideTranslations(locale, ["common", "huwelijksplanner-step-getuigen", "form"])),
   },
 });
 
-type WitnessType = {
-  name?: string;
-  email?: string;
+type Witness = {
+  name: string;
+  email: string;
+};
+
+type WitnessFormData = {
+  submit: "invite" | "continue";
+  witnesses?: Witness[];
 };
 
 export default function MultistepForm1() {
   const { t } = useTranslation(["common", "huwelijksplanner-step-getuigen", "form"]);
-  const data = { ...exampleState };
-  const { locale, push } = useRouter();
+  const [marriageOptions] = useContext(MarriageOptionsContext);
+  const { locale = "nl", push } = useRouter();
+  const formMethods = useForm<WitnessFormData>();
+  const [loading, setLoading] = useState(false);
+  const watch = formMethods.watch();
 
-  const [witness, setWitness] = useState<WitnessType>();
+  const hasWitnesses =
+    (watch.witnesses &&
+      watch.witnesses.some((witness) => {
+        return witness.name.length > 0 || witness.email?.length > 0;
+      })) ??
+    false;
 
-  const onWitnessSubmit = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    if (witness) {
-      push("/voorgenomen-huwelijk/getuigen/succes");
-    }
+  const onContinueWithoutInvite = () => {
+    push("/voorgenomen-huwelijk/getuigen/succes");
   };
 
-  const onWitnessChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setWitness({ ...witness, [event.target.name]: event.target.value });
-  };
+  const onWitnessSubmit = (formData: WitnessFormData) => {
+    if (!formData.witnesses) {
+      onContinueWithoutInvite();
+      return;
+    }
+
+    let hasError = false;
+
+    formData.witnesses.forEach((witness, index) => {
```

Comment:
I think these errors can already be added during the form-input registration. See comment below.